app/models/char_combat.rb

```diff
@@ -3,14 +3,20 @@ class CharCombat < ActiveRecord::Base
   belongs_to :first_combatant, :class_name => CharType.to_s
   belongs_to :second_combatant, :class_name => CharType.to_s
 
-  validates :first_combatant, :presence => true
-  validates :second_combatant, :presence => true
+  validates :first_combatant, :presence => true, :if => lambda { second_combatant == nil }
```

Comment:
You can write a function like `def smth . . . end` and pass it like this: `:if => :smth`.